### DISCUSSION TITLE
How to update the certificate currently in use?

### DISCUSSION THREAD

**Discussion by pass12335** [AdguardTeam/AdGuardHome#6162] [Q&A] [answered]
Hi all,

I am currently using Synology's NAS DS420+ and the Docker version of AdGuard Home. I use Synology's DDNS and have Let's Encrypt automatically generate the certificate. I use a script to copy these certificates to the AdGuard Home's read path every week, and I can see from the page that the related information has been updated.
However, recently when I open the AdGuard Home page in the browser, I am prompted that the certificate in use is about to expire. When will AdGuard Home update the certificate being used for the current connection?

![adgh](https://github.com/AdguardTeam/AdGuardHome/assets/9115315/0af57a88-7659-4a06-b0e7-649c7309f091)

AdGuard Home's current read path and the certificate information retrieved.
![adgh2](https://github.com/AdguardTeam/AdGuardHome/assets/9115315/3db5080b-3798-4ffa-a6c4-bcdebd50f859)

Cert info obtained when opening the AdGuard Home webpage in the Chrome browser.
![adgh3](https://github.com/AdguardTeam/AdGuardHome/assets/9115315/758bd1a6-9f6f-4cdb-a27a-1c90e057c250)

Thanks!

**Comment by ppfeufer** [comment, marked as answer]
AGH won't update the certificate, that's something you have to take care of.

In your case:
- Update the certificate
- Copy it over to the place where you need it for AGH
- Restart AGH and your web server that is used to proxy your AGH

**Comment by pass12335** [comment]
> AGH won't update the certificate, that's something you have to take care of.
> 
> In your case:
> 
> * Update the certificate
> * Copy it over to the place where you need it for AGH
> * Restart AGH and your web server that is used to proxy your AGH

Alright, it seems I need to add another script to periodically restart the AdGuard Home Docker container.
Thanks for the help!
